iterator.go

```diff
@@ -0,0 +1,267 @@
+			mi.db.mu.Unlock()
+		}
+	}()
+	for i := 0; i < len(mi.itrs); i++ {
+		if i == rank {
+			continue
+		}
+		itr := mi.itrsM[i]
+		if !itr.iter.Valid() {
+			continue
+		}
+		for itr.iter.Valid() &&
+			bytes.Equal(itr.iter.Key(), oldKey) {
+			if itr.rank > rank {
+				panic("rank error")
+			}
+			itr.iter.Next()
+		}
+		if !itr.iter.Valid() {
+			heap.Remove(&mi.h, itr.idx)
+		} else {
+			heap.Fix(&mi.h, itr.idx)
+		}
+	}
+}
+
+// Next moves the iterator to the next key.
+func (mi *Iterator) Next() {
+	if mi.h.Len() == 0 {
+		return
+	}
+
+	// Get the top item from the heap
+	topIter := mi.h[0]
+	mi.cleanKey(topIter.iter.Key(), topIter.rank)
+
+	// Check if the top item is a MemItr and a deleteKey
+	if topIter.iType == MemItr {
+		if valStruct, ok := topIter.iter.Value().(y.ValueStruct); ok && valStruct.Meta == LogRecordDeleted {
+			// Move to the next key and update the heap
+			topIter.iter.Next()
+			if !topIter.iter.Valid() {
+				heap.Remove(&mi.h, topIter.idx)
+			} else {
+				heap.Fix(&mi.h, topIter.idx)
+			}
+			// Call Next recursively to handle consecutive deleteKeys
+			mi.Next()
+		}
+	}
+
+	// Move to the next key and update the heap
+	topIter.iter.Next()
+	if !topIter.iter.Valid() {
+		heap.Remove(&mi.h, topIter.idx)
+	} else {
+		heap.Fix(&mi.h, topIter.idx)
+	}
+}
+
+// Key get the current key.
+func (mi *Iterator) Key() []byte {
+	return mi.h[0].iter.Key()
+}
+
+// Value get the current value.
+func (mi *Iterator) Value() []byte {
+	defer func() {
+		if r := recover(); r != nil {
+			mi.db.mu.Unlock()
+		}
+	}()
+	topIter := mi.h[0]
+	if topIter.iType == BptreeItr {
+		keyPos := new(KeyPosition)
+		keyPos.key = topIter.iter.Key()
+		keyPos.partition = uint32(mi.db.vlog.getKeyPartition(topIter.iter.Key()))
+		keyPos.position = wal.DecodeChunkPosition(topIter.iter.Value().([]byte))
+		record, err := mi.db.vlog.read(keyPos)
+		if err != nil {
+			panic(err)
+		}
+		return record.value
+	} else if topIter.iType == MemItr {
+		return topIter.iter.Value().(y.ValueStruct).Value
+	} else {
+		panic("iType not support")
+	}
+}
+
+// Valid returns whether the iterator is exhausted.
+func (mi *Iterator) Valid() bool {
+	if mi.h.Len() == 0 {
+		return false
+	}
+	topIter := mi.h[0]
+	if topIter.iType == MemItr && topIter.iter.Value().(y.ValueStruct).Meta == LogRecordDeleted {
+		mi.cleanKey(topIter.iter.Key(), topIter.rank)
+		topIter.iter.Next()
+		if topIter.iter.Valid() {
+			heap.Fix(&mi.h, topIter.idx)
+			return mi.Valid()
+		} else {
+			heap.Remove(&mi.h, topIter.idx)
+		}
+	} else if topIter.iType == BptreeItr && !topIter.iter.Valid() {
+		heap.Remove(&mi.h, topIter.idx)
+	}
+	return mi.h.Len() > 0
+}
+
+// Close the iterator.
+func (mi *Iterator) Close() error {
+	for _, itr := range mi.itrs {
+		err := itr.iter.Close()
+		if err != nil {
+			mi.db.mu.Unlock()
+			return err
+		}
+	}
+	mi.db.mu.Unlock()
+	return nil
+}
+
+func (db *DB) NewIterator(options IteratorOptions) (*Iterator, error) {
```

Comment:
Add more comments for this function to introduce this feature, and an usage example.